drivers/platform/altera/gpio.c

```diff
@@ -130,31 +131,7 @@ int32_t gpio_direction_input(struct gpio_desc *desc)
 int32_t gpio_direction_output(struct gpio_desc *desc,
```

Comment:
So this function is used in some drivers to set the gpio value?

Reply:
Generally speaking, the aim of this function is to set the direction of a gpio as output, and also the value of the gpio (high/low).

Since in this particular case there is no need to set the direction, the function will only set the value.

As a reference for a complete implementation for `gpio_direction_output`, you can have a look here: https://github.com/analogdevicesinc/no-OS/blob/5c6135e9d50b61527e184c429566c6488acc246c/drivers/platform/xilinx/gpio.c#L268-L279

`XGpio_SetDataDirection` sets the gpio direction, while  `XGpio_DiscreteWrite` sets the output value, both contained in the same function `gpio_direction_output`.

Reply:
I understand, I got confused the function with gpio_set_direction.